app/uk/gov/hmrc/customs/notification/repo/ClientNotificationRepo.scala

```diff
@@ -16,25 +16,91 @@
 
 package uk.gov.hmrc.customs.notification.repo
 
+import javax.inject.{Inject, Singleton}
+
+import play.api.libs.json.Json
+import reactivemongo.api.Cursor
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.BSONObjectID
+import reactivemongo.play.json.JsObjectDocumentWriter
 import uk.gov.hmrc.customs.notification.domain.{ClientNotification, ClientSubscriptionId}
+import uk.gov.hmrc.customs.notification.logging.NotificationLogger
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-/**
-  * Created by dev on 25/06/2018.
-  */
 trait ClientNotificationRepo {
 
+  def save(clientNotification: ClientNotification): Future[Boolean]
 
-  def save(cn: ClientNotification): Future[Boolean]
-  //FIFO based on whatever we decide to use, this method  has to return the list in insertion order. for now, leaving the timestamp in there but it yet to be decided.
-  // speak to Avinder & Paul to get more context
   def fetch(csid: ClientSubscriptionId): Future[List[ClientNotification]]
 
-
   def fetchDistinctNotificationCSIDsWhichAreNotLocked(): Future[Set[ClientSubscriptionId]]
 
-  //make sure we log it properly, we cant recover from delete failure, we might need to raise an alert for this one. We'll come back to this one.
-  def delete(mongoObjectId: String): Future[Unit]
+  def delete(mongoObjectId: BSONObjectID): Future[Unit]
```

Comment:
I have a vague recollection that for production we need a Write Concern that ensures we get an ACK from at least 2 nodes in the cluster. Just a note - maybe not for this PR

Reply:
Ok, will check this but not for this PR as you suggest.